src/ui/viewmodels/MessageBoxViewModel.hh

```diff
@@ -0,0 +1,167 @@
+#ifndef RA_UI_MESSAGEBOX_VIEW_MODEL_H
+#define RA_UI_MESSAGEBOX_VIEW_MODEL_H
+#pragma once
+
+#include "ui/WindowViewModelBase.hh"
+
+namespace ra {
+namespace ui {
+namespace viewmodels {
+
+enum class MessageBoxButtons
+{
+    OK,
+    OKCancel,
+    YesNo,
+    YesNoCancel,
+    RetryCancel,
+};
+
+enum class MessageBoxIcon
+{
+    None,
+    Info,
+    Warning,
+    Error,
+};
```

Comment:
I couldn't decide if these should be members of the `MessageBoxViewModel` (i.e. `MessageBoxViewModel::Icon` instead of `MessageBoxIcon`). I'm probably going to move them in, but wanted a second opinion.

Reply:
I think moving them in is fine.